e2e/gateway/config/config.go

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, NASD Inc. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN "AS IS" BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/BurntSushi/toml"
+)
+
+const ConfigDir = "./"
+
+// Config holds the complete configuration.
+type Config struct {
+	Ethereum EthereumConfig `toml:"ethereum"`
+}
+
+// Load loads configuration from a TOML file.
+func Load(path string) (*Config, error) {
+	var cfg Config
+
+	if _, err := toml.DecodeFile(path, &cfg); err != nil {
+		return nil, fmt.Errorf("failed to decode config file: %w", err)
+	}
+
+	if err := cfg.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	return &cfg, nil
+}
+
+// Validate validates the configuration.
+func (c *Config) Validate() error {
+	if err := c.Ethereum.Validate(); err != nil {
+		return fmt.Errorf("ethereum configuration error: %w", err)
+	}
+
+	return nil
+}
```

Comment:
could add the nil check here too


```suggestion
func (c *Config) Validate() error {
	if c == nil {
		return errors.New("nil config")
	}
	
	if err := c.Ethereum.Validate(); err != nil {
		return fmt.Errorf("ethereum configuration error: %w", err)
	}

	return nil
}
```